data/spells/lib/spells.lua

```diff
@@ -296,3 +306,74 @@ function Creature:addDamageCondition(target, conditionType, listType, damage, ti
 	target:addCondition(condition)
 	return true
 end
+
+function isOnCombatArea(casterPosition, targetPosition, coordinates)
+	if casterPosition.z == targetPosition.z then
+		for _, axis in ipairs(coordinates) do
+			if casterPosition.x + axis.x == targetPosition.x and casterPosition.y + axis.y == targetPosition.y then
+				return true
+			elseif casterPosition.x - axis.x == targetPosition.x and casterPosition.y + axis.y == targetPosition.y then
+				return true
+			elseif casterPosition.x - axis.x == targetPosition.x and casterPosition.y - axis.y == targetPosition.y then
+				return true
+			elseif casterPosition.x + axis.x == targetPosition.x and casterPosition.y - axis.y == targetPosition.y then
+				return true
+			end
+		end
+	end
+	return false
+end
```

Comment:
This is just a horrible way to use plain old areas, literally. You could iterate over the spell area instead. I don't understand why you changed this.